arbnode/espresso_transaction_queue.go

```diff
@@ -0,0 +1,117 @@
+package arbnode
+
+import (
+	"context"
+	"slices"
+	"sync"
+	"time"
+
+	espressoClient "github.com/EspressoSystems/espresso-sequencer-go/client"
+	espressoTypes "github.com/EspressoSystems/espresso-sequencer-go/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	flag "github.com/spf13/pflag"
+)
+
+type EspressoTransactionQueue struct {
+	stopwaiter.StopWaiter
+
+	config EspressoTransactionQueueConfigFetcher
+
+	espressoClient *espressoClient.Client
+	newTxnNotifier chan struct{}
+
+	pendingTxnsQueueMutex sync.Mutex // cannot be acquired while reorgMutex is held
+	pendingTxnsQueue      []*espressoTypes.Transaction
+
+	submittedTxn *espressoTypes.Transaction
+}
+
+type EspressoTransactionQueueConfig struct {
+	HotShotUrl                  string        `koanf:"hotshot-url"`
+	EspressoNamespace           uint64        `koanf:"espresso-namespace"`
+	EspressoTxnsPollingInterval time.Duration `koanf:"espresso-polling-interval"`
+}
+
+type EspressoTransactionQueueConfigFetcher func() *EspressoTransactionQueueConfig
+
+var DefaultEspressoTransactionQueueConfig = EspressoTransactionQueueConfig{
+	HotShotUrl:                  "",
+	EspressoTxnsPollingInterval: time.Millisecond * 100,
+}
+
+var TestEspressoTransactionQueueConfig = EspressoTransactionQueueConfig{
+	HotShotUrl:                  "",
+	EspressoTxnsPollingInterval: time.Millisecond,
+}
+
+func EspressoTransactionQueueConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.String(prefix+".hotshot-url", DefaultEspressoTransactionQueueConfig.HotShotUrl, "url of the hotshot sequencer")
+	f.Uint64(prefix+".espresso-namespace", DefaultEspressoTransactionQueueConfig.EspressoNamespace, "espresso namespace that corresponds the L2 chain")
+	f.Duration(prefix+".espresso-polling-interval", DefaultEspressoTransactionQueueConfig.EspressoTxnsPollingInterval, "interval between polling for transactions to be included in the block")
+}
+
```

Comment:
From your implemention, this transaction queue is specified from the transaction streamer config. So it is unnecessary to add flags for this.